lib/cardano_explorer/transaction.rb

```diff
@@ -1,6 +1,10 @@
 module CardanoExplorer
   class Transaction < Base
 
+    def latest
+      get_and_parse 'last'
+    end
+
```

Comment:
At first, I wanted to make this a class method but ran into issues with private methods in Base. 

Perhaps we should revamp the method type of `find` in Base class? Changing `find` from class method to instance method would play nicer with my addition of #latest and further additions to all classes that don't depend on an id. 


```ruby
transactions = CardanoExplorer::Transaction.new
transactions.latest
transaction =  transactions.find(id)
transaction.summary
```

The new pattern also works well with the other classes.

Reply:
So the `api/txs/last` returns an array of transactions ids (and a couple other fields):
```json
{
   "Right":[
      {
         "cteId":"d5efca4b13c5001c8cbc5617523a7562d6746ff6e454887ba28205e22c854d32",
         "cteTimeIssued":1516154031,
         "cteAmount":{
            "getCoin":"548416093154"
         }
      },
      {
         "cteId":"a3d4531f8955e69b922bee29b1cb615ebcb7d48aba0d1fd132553b9820c30869",
         "cteTimeIssued":1516153991,
         "cteAmount":{
            "getCoin":"548441342333"
         }
      },
      ...
   ]
}
```

Inside the "Right" key in that JSON is an array of transaction objects. So I think it would be more representative if we make it so that `Transaction.latest` returns an Array of instantiated transaction objects.

For example this would allow the users to be able to do things like this:
```ruby
transactions = Transaction.latest #=>
# [#<CardanoExplorer::Transaction ...>, #<CardanoExplorer::Transaction ...>, #<CardanoExplorer::Transaction ...>]

# Then, the user can do things like:
transactions.each { |transaction| .... }
transactions.last.summary #=> { ... }
transactions.first.summary #=> { ... }
transactions[3].summary #=> { ... }
```

I know right now the Base class doesn't support that, and it was my fault for defining a Base class so early on in the development of this gem. However, we can make changes to the Base class to support this.

First what we can do is make the HTTP request related methods to become a class method like so:
```
  class Base

   # Other code

  private

    def self.get(path, options = {})
      HTTParty.get "#{root_url}/#{endpoint}/#{path}", query: options
    end

    def self.root_url
      CardanoExplorer.configuration.root_url
    end

    def self.endpoint
      raise NotImplementedError, '#endpoint not defined in child class'
    end

    def get(path, options = {})
      self.class.get path, options
    end
  end
```

In Transaction class we can add this to return an Array of instantiated Transaction objects:
``` ruby
    def self.latest
      get('last').parsed_response['Right'].map {|txs| new txs['cteId']}
    end
```

We also need to make sure that the `endpoint` method in all the other classes become a class method as well.
```
  private

    def self.endpoint
      # whatever their respective endpoints are.
    end
```

Sorry for such a big comment. What do you or anyone else think of that approach?

Reply:
Having `Transaction.latest` returns an Array of instantiated transaction objects is a great idea.